commit 9be5e6870224b7e3214716e51381568ed5eecda8 (refs/changes/47/3247/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2018-01-23T09:46:16-08:00 (1 year, 8 months ago)
    
    TRITON-66 'sdcadm experimental update-other' crashes on a non-master/non-release SAPI image version
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/lib/steps/sapi.js b/lib/steps/sapi.js
index 74003f3..315d7f6 100644
--- a/lib/steps/sapi.js
+++ b/lib/steps/sapi.js
@@ -144,13 +144,21 @@ function sapiEnsureAgentServices(arg, cb) {
                     MIN_VALID_SAPI_VERSION;
             } else if (splitVersion[0] === 'release') {
                 validSapi = splitVersion[1] >= MIN_VALID_SAPI_VERSION;
+            } else {
+                progress('Warning: cannot verify that SAPI is at least of ' +
+                    '%s vintage, because a non-master/non-release SAPI ' +
+                    'image is being used.', MIN_VALID_SAPI_VERSION);
+                validSapi = true;
             }
 
             if (!validSapi) {
-                return next(new errors.SDCClientError(new Error('Datacenter ' +
-                    'does not have the minimum SAPI version needed for adding' +
-                    ' service agents. ' +
-                    'Please try again after upgrading SAPI', 'sapi')));
+                return next(new errors.SDCClientError(
+                    new Error(util.format('SAPI in this datacenter is using ' +
+                        'image version %s, which is not the minimum version ' +
+                        'required for adding service agents (%s). Please ' +
+                        'upgrade SAPI and then retry.', img.version,
+                        MIN_VALID_SAPI_VERSION)),
+                    'sapi'));
             }
 
             return next();
